dali/test/python/test_operator_readers_webdataset_base.py

```diff
@@ -0,0 +1,112 @@
+import nvidia.dali as dali
+from nvidia.dali import pipeline_def
+import nvidia.dali.fn as fn
+import nvidia.dali.fn.readers as readers
+from test_utils import compare_pipelines, get_dali_extra_path
+from nose.tools import assert_raises, assert_equal
+import tempfile
+from subprocess import call
+import os
+from glob import glob
+import tarfile
+import numpy as np
+import math
+
+test_batch_size = 4
+wds2idx_script = "../../../tools/wds2idx.py"
+
+
+@pipeline_def()
+def webdataset_raw_pipeline(
+    uris,
+    configs,
+    ext,
+    missing_component_behavior="fillempty",
+    dtypes=None,
+    dont_use_mmap=False,
+    num_shards=1,
+    shard_id=0,
+    skip_cached_images=False,
+    pad_last_batch=False,
+    lazy_init=False,
+    read_ahead=False,
+    stick_to_shard=False,
+):
+    out = readers.webdataset(
+        uris=uris,
+        configs=configs,
+        ext=ext,
+        missing_component_behavior=missing_component_behavior,
+        dtypes=dtypes,
+        dont_use_mmap=dont_use_mmap,
+        prefetch_queue_depth=1,
+        num_shards=num_shards,
+        shard_id=shard_id,
+        stick_to_shard=stick_to_shard,
+        skip_cached_images=skip_cached_images,
+        pad_last_batch=pad_last_batch,
+        lazy_init=lazy_init,
+        read_ahead=read_ahead,
+    )
+    return out if type(out) != list else tuple(out)
+
+
+def filter_ext(files, exts):
+    if type(exts) == str:
+        exts = {exts}
+    return list(filter(lambda s: any(map(lambda ext: s.endswith("." + ext), exts)), files))
+
+
+@pipeline_def()
+def file_reader_pipeline(
+    files,
+    exts=None,
+    dont_use_mmap=False,
+    num_shards=1,
+    shard_id=0,
+    skip_cached_images=False,
+    pad_last_batch=False,
+    lazy_init=False,
+    read_ahead=False,
+    stick_to_shard=False,
+):
+    if type(exts) != list:
+        exts = [exts]
```

Comment:
```suggestion
   exts = list(exts)
```
When exts is list this is no op.